v0/destinations/posthog/transform.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-undef */
+/* eslint-disable import/no-unresolved */
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  constructPayload,
+  defaultPostRequestConfig
+} = require("../../util");
+
+// Logic To match destination Property key that is in Rudder Stack Properties Object.
+const generatePropertyDefination = message => {
+  const PHPropertyJson = [
+    { destKey: "$os", sourceKeys: "context.os.name" },
```

Comment:
need to review the sourceKeys.
ex: 

- os/browser from JS source are not readily available, we have to parse the UA.
- $initial_referrer/$initial_referring_domain can be skipped
- ip to come from [request_ip, context.ip] ?
- remove distinct_id ? 
- fill in other Todos

Reply:
Fixed all except 2 To Do.
Left To Do's are portion of code that won't break anything but to make code much readable  .